### PR TITLE
machines: New "Disks" subtab added

### PR DESCRIPTION
**Pull request by mareklibra**
VM detail is enriched for new "Disks" subtab listing all disks assigned to particular VM.

 * [x] Video or screenshot

Replaces https://github.com/cockpit-project/cockpit/pull/6019, https://github.com/cockpit-project/cockpit/pull/6067

### CODE REVIEW

File: pkg/machines/vmdiskstab.jsx [hunk 1]
```diff
@@ -0,0 +1,116 @@
+/*jshint esversion: 6 */
```
Comment (martinpitt) [inline pkg/machines/vmdiskstab.jsx]: This adds an entire new file which isn't related to the fix in cockpit-components-listing.jsx nor documented in the commit message. This should probably be split out into a separate commit or even PR, as it doesn't even look related to "fix domstats"?

File: pkg/machines/config.es6 [hunk 2]
```diff
@@ -35,7 +35,7 @@ const VMS_CONFIG = {
         }
     },
     // TODO: make it configurable via config file
-    isDev: false, // Never commit with 'true'
+    isDev: true, // Never commit with 'true'
```
Comment (martinpitt) [inline pkg/machines/config.es6]: This contradicts the comment, so this should be reverted, or the comment updated with a justification.

File: pkg/machines/vmdiskstab.jsx (hunk 1 above)
Comment (dperpeet) [inline pkg/machines/vmdiskstab.jsx]: good catch, this change should be part of https://github.com/cockpit-project/cockpit/pull/6072/commits/1d775591b41dd5b23cdf20376ffd9f3e50905eb8

File: pkg/machines/config.es6 (hunk 2 above)
Comment (dperpeet) [inline pkg/machines/config.es6]: if I understand correctly this is part of debugging the issue at hand

**Comment by mareklibra** [comment]
@martinpitt , right, this is so far debugging. I'll let you know once ready. I started to run the tests locally

**Comment by mareklibra** [comment]
Let's wait for test results, but disk statistics should be fixed now for debian.
Disk allocation statistics have been added to Libvirt since version 1.2.18.

So far, newer Libvirt version is not required for remaining functionality of the plugin. So let's not add requirement on >1.2.18, it would be too restrictive.

With recent implementation, if disk capacity/allocation is not available, the columns remains empty. Any suggestion how/if this state should be communicated to the user?

If disk statistics is required on old systems, it would need to be gathered differently. Not an easy task considering so many possible disk sources.

**Comment by dperpeet** [comment]
> So far, newer Libvirt version is not required for the rest of cockpit:machines functionality. So let's not add requirement on >1.2.18, it would be too restrictive just because of this feature.

I agree, let's not be too restrictive.

> With recent implementation, if disk capacity/allocation is not available, the columns remain blank. Any suggestion how/if this state should be communicated to the user? Like a "triangle" icon with a tooltip in the header column?

@garrett @andreasn Can we just remove the column in that case? I don't think we need to warn the user, it's not something that went wrong. Another way could be to add some text instead, along the lines of "libvirt too old for statistics", but that might be crowding the UI.

> If disk statistics are required on old systems, it would need to be gathered differently. Not an easy task considering so many possible disk sources.

The library should be updated at some point in the future. I don't think a workaround is necessary.

**Comment by dperpeet** [comment]
The changes itself look good, but there is the matter of the changes in the wrong commit to be resolved.

**Comment by mareklibra** [comment]
Removing columns if data can't be retrieved for them make sense as well. And maybe it will look better - it's just not supported.
On the other hand - the proposed tooltip (or whatever similar) might hold information what the user can do to fix it (like upgrade Libvirt)

**Comment by dperpeet** [comment]
> Removing columns if data can't be retrieved for them make sense as well. And maybe it will look better - it's just not supported.
> On the other hand - the proposed tooltip (or whatever similar) might hold information what the user can do to fix it (like upgrade Libvirt)

Right. We did something similar for SELinux, when setroubleshoot-server isn't installed: https://github.com/cockpit-project/cockpit/pull/5868#issuecomment-279454986

File: pkg/machines/vmdiskstab.jsx (hunk 1 above)
Comment (mareklibra) [inline pkg/machines/vmdiskstab.jsx]: Done

**Comment by mareklibra** [comment]
Nice. So if no objections, I'm going to do it similarly.

**Comment by stefwalter** [comment]
@mareklibra Could you record a video or screenshot (if this feature is just a static display) for the release notes? There's tips for video recording in HACKING.md

**Comment by mareklibra** [comment]
@stefwalter , the feature is more or less static, so what about this screenshot?
disks

**Comment by mareklibra** [comment]
For completeness, this is how the subtab is rendered for old libvirt version:
vmdiskstatsoldlibvirt

**Comment by dperpeet** [comment]
the screenshots are good, thanks

**Comment by dperpeet** [comment]
remaining failures are unrelated
